spec/unit/settings_spec.rb

```diff
@@ -221,6 +221,27 @@ def metadata(setting)
       @settings.define_settings :main, :bool => { :type => :boolean, :default => true, :desc => "desc" }
     end
 
+    it "the cache should be threadsafe" do
+      original_cache = @settings.instance_variable_get(:@cache).value
+      other_request = Thread.new do
+        expect(@settings.instance_variable_get(:@cache).value).not_to be(original_cache)
+      end
+      other_request.join
```

Comment:
What is the purpose of this?

Reply:
This will block until the thread it's called on completes.

Reply:
damn turns out when you don't feel rushed and actually take the time to google things, you can learn a lot: https://www.rubyguides.com/2015/07/ruby-threads/ 😅 

thank you for answering all my questions, I really appreciate it